v3/lints/rfc/lint_subject_given_name_max_length.go

```diff
@@ -52,11 +70,20 @@ func (l *subjectGivenNameMaxLength) CheckApplies(c *x509.Certificate) bool {
 }
 
 func (l *subjectGivenNameMaxLength) Execute(c *x509.Certificate) *lint.LintResult {
-	for _, j := range c.Subject.GivenName {
-		if utf8.RuneCountInString(j) > 16 {
+	for _, givenName := range c.Subject.GivenName {
+		characters := utf8.RuneCountInString(givenName)
+		if characters > 32768 {
 			return &lint.LintResult{Status: lint.Error}
+		} else if characters > 64 {
+			return &lint.LintResult{
+				Details: fmt.Sprintf("Although RFC 5280 prescribes a maximum "+
+					"length of 32768, the 1988 ITU-T for x.509 prescribes a maximum "+
+					"length of 64 characters. As such, your %d character long given "+
+					"name may encounter interoperability issues with systems that "+
+					"were built targeting x.509's 64 character limit.", characters),
```

Comment:
```suggestion
				Details: fmt.Sprintf("givenName is %d characters long. givenNames that are longer "+
				        "than 64 characters may not interoperate or decode properly.", characters),
```

I'm not fundamentally opposed to more context, but that seems like a considerable amount. While I'm requesting (below) to move this into a separate lint, I think we could link to some of the context/discussion in the code, since I think we assume a CA will read the lint if/when they get a warning from it? :) Or at least the description/citation, where we can also provide context :)